libafl_bolts/src/lib.rs

```diff
@@ -1050,6 +1050,46 @@ pub unsafe fn set_error_print_panic_hook(new_stderr: RawFd) {
     }));
 }
 
+// Credit goes to https://github.com/thomcc/nonzero_lit
```

Comment:
I don't think that all this complexity is needed, you can just do
```rust
use std::num::NonZeroUsize;
const nz: NonZeroUsize = match NonZeroUsize::new(4) {
    Some(x) => x,
    None => panic!("was zero")
};
```
This has been available since 2021: https://github.com/rust-lang/rust/pull/89508

Reply:
`const Option::unwrap` just landed, too: https://github.com/rust-lang/rust/issues/67441

Reply:
That's what we use currently but it's more ugly.
Maybe we could just do a 
```rust
const fn nonzero(val: usize) -> NonZeroUsize {
    NonZero::new(val).unwrap()
}
```
?

Reply:
> Maybe we could just do a

this is not compile time check

Reply:
We want to use macro because i don't want to see bunches of unwrap() inside our lib

Reply:
I understand the motivation for a macro, I just think this macro could be implemented in a simpler way: https://github.com/AFLplusplus/LibAFL/pull/2624